modules/core/src/math/matrix/vpMatrix.cpp

```diff
@@ -1639,10 +1630,7 @@ void vpMatrix::kron(const vpMatrix &m1, const vpMatrix &m2, vpMatrix &out)
   unsigned int r2 = m2.getRows();
   unsigned int c2 = m2.getCols();
 
-  if (r1 * r2 != out.rowNum || c1 * c2 != out.colNum) {
-    vpERROR_TRACE("Kronecker prodect bad dimension of output vpMatrix");
-    throw(vpException(vpException::dimensionError, "In Kronecker product bad dimension of output matrix"));
-  }
+  out.resize(r1*r2, c1*c2, false, false);
```

Comment:
New code will automatically resize the output matrix whereas before the output matrix must be created with the correct size, otherwise it throws an exception.
I don't think it is an issue but in case.

Reply:
I agree with you